CONTRIBUTING.md

```diff
@@ -118,28 +118,81 @@ The development process thus far has involved maintaining separate branches
-then merge to master and repeat for subsequent branches. When all the things
-planned for release have been merged to master, create a version bump commit.
-I've used `npm version major.minor.patch` for this, but it can be done manually
-if you prefer.
+`env.sh` defines environment variables for the testing modules; it must be sourced,
+not invoked as an executable. To setup the environment for the `npm test` command run
 
-After the version bump commit has been made, make sure it is tagged, committed, and pushed. Then
-be sure to push the tag using `git push origin <tag-name>`. If
-you just `git push` the tag will not be pushed. Note that `npm version` creates the
-tag in git; you don't need to create it manually.
 
-After all commits and tags have been pushed to git, it's simply a matter of
-running `npm publish` to send the latest version to the npm registry.
+`. env.sh bash`.
+
+The primary documentation for `env.sh` is the file itself.
+
+After `. env.sh bash` you should be able to run the test suite using `npm test`. You can
+also use gulp directly to choose specific tests like `gulp test:unit` or `gulp test:probes`
+or even a single specific test with `gulp test:probe:generic-pool`.
+
+### Releasing
 
+When you are ready to release, create a staging branch (name n.n.n - the intended version
+of the release) from master, rebase your branch(es) off the staging branch, run the local
+tests, and repeat for any additional branch(es). If there is only one branch then it may
+be used instead of creating a staging branch.
+
+When all items planned for the release have been incorporated into the staging branch then
+run additional some additional tests. At a minimum, start with a clean copy of the repository
+and run the test suite again. This will catch missing dependencies that may still be in the
+`node_modules` directory in a development area. So, in some directory,
+
+```
+git clone --depth=1 https://github.com/appoptics/appoptics-apm-node clean-apm
+cd clean-apm
+git checkout n.n.n     # the branch name
+npm install
+npm test               # presumes the docker environment for testing (see above).
+```
+
+If the change is significant you may need to run the support matrix again. That's beyond
+the scope of this document but involves testing appoptics against each released version
+of each package.
+
+An an extra precaution it is useful to test this in a real-world test harness. I use
+https://github.com/bmacnaughton/todomvc-mongodb. In the the `clean-apm` directory use
+`npm pack` to make a `.tgz` file, copy that to the `todomvc-mongodb` directory, change
+`package.json` for the `appoptics-apm` dependency to reference the `.tgz` file, install
+using `npm install`, setup the environment with `. env.sh stg` (works against the staging
+server - you might use another key), and run the server with `node server.js --fe_ip=localhost:port`.
+Use curl or a browser to execute requests against the server then check the backend to make
```

Comment:
Is it clearer to say `then check the AppOptics dashboard to make sure...`

Reply:
exactly why I wanted a review; I didn't have the right term in mind. much better - thanks.